### PR TITLE
Rename Type.IsSzArray to Type.IsSZArray and make public.

### PR DESCRIPTION
**Pull request by JonHanna**
For dotnet/corefx#16566

### CODE REVIEW

File: src/mscorlib/src/System/RuntimeHandles.cs [hunk 1]
```diff
@@ -153,7 +153,7 @@ internal static bool IsArray(RuntimeType type)
             return (corElemType == CorElementType.Array || corElemType == CorElementType.SzArray);
         }
 
-        internal static bool IsSzArray(RuntimeType type)
+        internal static bool IsSZArray(RuntimeType type)
```
Comment (JonHanna) [inline src/mscorlib/src/System/RuntimeHandles.cs]: This rename isn't necessary, but the consistency seemed worthwhile.

File: src/mscorlib/src/System/Type.cs [hunk 2]
```diff
@@ -1182,7 +1182,7 @@ public bool IsArray
             get { return IsArrayImpl(); }
         }
 
-        internal virtual bool IsSzArray
+        public virtual bool IsSZArray
         {
             [Pure]
             get { return false; }
```
Comment (AtsushiKan) [inline src/mscorlib/src/System/Type.cs]: @JonHanna  The default implemention now needs to be "throw new NotImplementedException();"

**Comment by AtsushiKan** [comment]
Because the default implementation of IsSZArray is now to throw NYI, any corelib types that derive from Type may need to override this now to preserve the existing behavior. System.Reflection.Emit.TypeBuilder comes to mind.

File: src/mscorlib/src/System/Type.cs (hunk 2 above)
Comment (JonHanna) [inline src/mscorlib/src/System/Type.cs]: Why can't we just keep the default false return?

Comment (AtsushiKan) [inline src/mscorlib/src/System/Type.cs]: The API was approved based on the "throw NotImplemented" behavior. If a third party type fails to override this, we want to get a clear exception to that effect, not an assumed answer that will be wrong 10% of the time.

**Comment by JonHanna** [comment]
Test Windows_NT x64 Release Priority 1 Build and Test

**Comment by JonHanna** [comment]
Test Windows_NT x64 Release Priority 1 Build and Test

**Pull request merged**